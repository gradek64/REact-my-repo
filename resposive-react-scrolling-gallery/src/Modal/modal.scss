.clickAway {
  width: 100vw;
  height: 100vh;
  position: fixed;
  display: flex;
  top: 0;
  left: 0;
}

.modal {
  width: 100vw;
  height: 100vh;
  position: fixed;
  display: flex;
  visibility: visible;
  align-items: center;
  justify-content: center;
  top: 0;
  left: 0;
  background-color: rgba(0, 0, 0, 0.5);
  z-index: 100;
  opacity: 1;

  /*you need animate most of the properties all is just a cheat to get animation back on class remove;*/
  transition: all 1s;

  &.close-state {
    /*for animation you need to use visibility not display property*/
    visibility: hidden;
    opacity: 0;
  }

  .modal-content {
    padding: 2rem;

    /*width: 80%;*/
    max-width: 1000px;
    background-color: white;
    position: absolute;
  }

  .modal-content {
    transition: right 1s;
    right: -100%;

    &.slide-in {
      right: 0;
    }
  }

  &.no-animation {
    .modal-content {
      right: 0;
    }
  }

  .close-modal {
    position: absolute;
    max-width: 100px;
    width: 5%;
    height: auto;
    top: -1rem;
    border-radius: 50%;
    border: 1px solid;
    cursor: pointer;
    right: -1rem;
    font-size: 3rem;
  }
}