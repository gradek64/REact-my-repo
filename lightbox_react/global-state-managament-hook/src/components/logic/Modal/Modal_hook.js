import React, { useRef } from 'react';
import PropTypes from 'prop-types';
import dispatchEvent from '../../../customEvents/dispatchEvent';
import useStateGlobal from '../../global_state_hook/global_hook_state';

// custom imports;
import './modal.scss';

const Modal = (props) => {
  const modalDomElement = useRef(null);
  /**
   *
   * it only way to re-render a component is either using hook or class
   * component, thefore we register a hook here that is using
   * higher order hook that returns useState()
   * this internal/global hook updates global state
   * thefore affecting every component connected to useStateGlobal()
   * this causing rerending component connected to useStateGlobal()
   *
   **/
  const [state, updateStateFromGlobal] = useStateGlobal();

  //------- internal close method ----
  const { modalID: modalIDprop } = props;
  const closeModal = (e) => {
    if (e && e.target !== e.currentTarget) return;
    if (e.type === 'keydown' && e.key !== 'Escape') return;
    updateStateFromGlobal({ open: false, modalID: modalIDprop });
  };
  //------- ------- ----------

  const { open, modalID } = state;
  const openCurrent = open && modalID === modalIDprop;
  if (openCurrent)
    setTimeout(() => {
      modalDomElement.current.focus();
    }, 100);

  console.log('state in modal', state);
  //this is for preventing uneccessary render
  if (!openCurrent) return null;
  console.log('re-rendered');
  return (
    <React.Fragment>
      <div
        data-testid="modal"
        style={openCurrent ? { display: 'flex' } : { display: 'none' }}
        tabIndex={openCurrent ? -1 : 0}
        role="presentation"
        className="modal"
        //this is currentTarget
        onClick={closeModal}
        onKeyDown={closeModal}
        ref={modalDomElement}
      >
        <div className="modal-content">{props.children}</div>
      </div>
    </React.Fragment>
  );
};

Modal.propTypes = {
  modalID: PropTypes.string.isRequired,
  children: PropTypes.any.isRequired,
};
export default Modal;
