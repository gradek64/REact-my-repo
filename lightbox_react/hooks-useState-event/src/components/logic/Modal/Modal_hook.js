import React, { useState } from 'react';
import PropTypes from 'prop-types';

// custom imports;
import dispatchEvent from '../../../customEvents/dispatchEvent';
import './modal.scss';

const Modal = (props) => {
  //open state is an invisible function that only task is to manage open state!
  //call it everwhere is needed;
  const [open, openState] = useState(false);

  //------- internal open and close method ----
  const openModal = () => {
    const delaySelector = setTimeout(() => {
      clearTimeout(delaySelector);
      document.querySelector('[tabIndex="-1"]').focus();
    }, 100);
    openState(true);
  };
  const closeModal = (e) => {
    if (e && e.target !== e.currentTarget) return;
    if (e.type === 'keydown' && e.key !== 'Escape') return;
    openState(false);
  };
  //------- ------- ----------

  //register event with custom event
  const { eventOn } = props;
  dispatchEvent.on(eventOn, openModal);

  return (
    <React.Fragment>
      <div
        data-testid="modal"
        style={open ? { display: 'flex' } : { display: 'none' }}
        tabIndex={open ? -1 : 0}
        role="presentation"
        className="modal"
        //this is currentTarget
        onClick={closeModal}
        onKeyDown={closeModal}
      >
        <div className="modal-content">{props.children}</div>
      </div>
    </React.Fragment>
  );
};

Modal.propTypes = {
  eventOn: PropTypes.string.isRequired,
  children: PropTypes.any.isRequired,
};
export default Modal;
