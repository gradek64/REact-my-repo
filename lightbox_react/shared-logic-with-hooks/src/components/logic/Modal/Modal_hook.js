import React, { useState, useEffect } from 'react';
import PropTypes from 'prop-types';

// custom imports;
import customEvent from '../../../customEvents/dispatchEvent';
import './modal.scss';

const Modal = (props) => {
  //open state is an invisible function that only task is to manage open state!
  //call it everwhere is needed;
  const [open, openState] = useState(false);

  //------- internal open and close method ----
  const openModal = () => {
    const delaySelector = setTimeout(() => {
      clearTimeout(delaySelector);
      document.querySelector('[tabIndex="-1"]').focus();
    }, 100);
    openState(true);
  };
  const closeModal = (e) => {
    if (e && e.target !== e.currentTarget) return;
    if (e.type === 'keydown' && e.key !== 'Escape') return;
    openState(false);
  };
  //------- ------- ----------

  useEffect(() => {
    const { eventOn } = props;

    if (!eventOn) {
      console.warn('your custom event is not specified');
      return;
    }
    customEvent.on(eventOn, openModal);
    customEvent.on(`CLOSE_${eventOn}`, closeModal);
  }, []); // array of conditions is empty so run only once is mounted

  return (
    <React.Fragment>
      <div
        style={open ? { display: 'flex' } : { display: 'none' }}
        tabIndex={open ? -1 : 0}
        role="presentation"
        className="modal"
        //this is currentTarget
        onClick={closeModal}
        onKeyDown={closeModal}
      >
        <div className="modal-content">{props.children}</div>
      </div>
    </React.Fragment>
  );
};

Modal.propTypes = {
  eventOn: PropTypes.string.isRequired,
  children: PropTypes.any.isRequired,
};
export default Modal;
