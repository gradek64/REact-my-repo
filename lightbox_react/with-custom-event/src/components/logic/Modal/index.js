import React from 'react';
import PropTypes from 'prop-types';

// custom imports;
import customEvent from '../../../customEvents/dispatchEvent';
import './modal.scss';

class Modal extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      open: false,
    };
    this.openModal = this.openModal.bind(this);
    this.closeModal = this.closeModal.bind(this);
  }

  componentDidMount = () => {
    const { eventOn } = this.props;

    if (!eventOn) {
      console.warn('your custom event is not specified');
      return;
    }
    customEvent.on(eventOn, this.openModal);
    customEvent.on(`CLOSE_${eventOn}`, this.closeModal);
  };

  componentWillUnmount() {
    // only when you change route though;
    // remove custom Event
  }

  openModal() {
    const delaySelector = setTimeout(() => {
      clearTimeout(delaySelector);
      document.querySelector('[tabIndex="-1"]').focus();
    }, 100);
    this.setState({ open: true });
  }
  closeModal(e) {
    if (e && e.target !== e.currentTarget) return;
    if (e.type === 'keydown' && e.key !== 'Escape') return;
    this.setState({ open: false });
  }

  render() {
    const { open } = this.state;
    return (
      <React.Fragment>
        <div
          style={open ? { display: 'flex' } : { display: 'none' }}
          tabIndex={open ? -1 : 0}
          role="presentation"
          className="modal"
          //this is currentTarget
          onClick={this.closeModal}
          onKeyDown={this.closeModal}
          ref={this.clickAway}
        >
          <div className="modal-content">{this.props.children}</div>
        </div>
      </React.Fragment>
    );
  }
}

Modal.propTypes = {
  eventOn: PropTypes.string.isRequired,
  children: PropTypes.any.isRequired,
};
export default Modal;
