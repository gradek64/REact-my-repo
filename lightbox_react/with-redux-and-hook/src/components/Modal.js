import React, { useRef } from 'react';
import { useDispatch, useSelector, shallowEqual } from 'react-redux';
import PropTypes from 'prop-types';
import { closeModal } from '../actions';

// custom imports;
import '../components/modal.scss';

const Modal = (ownprops) => {
  const modalDomElement = useRef(null);

  const { modalID: ownID } = ownprops;

  const { open, modalID } = useSelector(
    (state) => ({
      open: state.modalReducer.open,
      modalID: state.modalReducer.modalID,
    }),
    shallowEqual //this allows to update property if neeeded by shallow comparison of value
  );
  const dispatch = useDispatch();

  const openCurrent = open && modalID === ownID;
  if (openCurrent)
    setTimeout(() => {
      modalDomElement.current.focus();
    }, 100);

  const closeModalInternal = (e) => {
    if (e && e.target !== e.currentTarget) return;
    if (e.type === 'keydown' && e.key !== 'Escape') return;
    dispatch(closeModal());
  };

  return (
    <React.Fragment>
      <div
        style={openCurrent ? { display: 'flex' } : { display: 'none' }}
        tabIndex={openCurrent ? -1 : 0}
        role="presentation"
        className="modal"
        //this is currentTarget
        onClick={closeModalInternal}
        onKeyDown={closeModalInternal}
        ref={modalDomElement}
      >
        <div className="modal-content">{ownprops.children}</div>
      </div>
    </React.Fragment>
  );
};

Modal.propTypes = {
  children: PropTypes.any.isRequired,
};

export default Modal;

/* const mapStateToProps = (state, props) => {
  const { open, modalID } = state.modalReducer;
  const { modalID: ownID } = props;
  return {
    open,
    modalID,
    ownID,
  };
};

const mapDispatchToProps = {
  closeModal,
};

export default connect(mapStateToProps, mapDispatchToProps)(Modal); */
