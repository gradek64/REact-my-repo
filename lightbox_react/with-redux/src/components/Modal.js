import React from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import { closeModal } from '../actions';

// custom imports;
import '../components/modal.scss';

class Modal extends React.Component {
  modalDomElement = React.createRef();

  componentWillUnmount() {
    // only when you change route though;
    // remove custom Event
  }

  closeModal = (e) => {
    const { closeModal: closeModalGlobal } = this.props;
    if (e && e.target !== e.currentTarget) return;
    if (e.type === 'keydown' && e.key !== 'Escape') return;
    closeModalGlobal();
  };

  render() {
    const { open, modalID, ownID } = this.props;
    const openCurrent = open && modalID === ownID;
    if (openCurrent)
      setTimeout(() => {
        this.modalDomElement.current.focus();
      }, 100);
    return (
      <React.Fragment>
        <div
          style={openCurrent ? { display: 'flex' } : { display: 'none' }}
          tabIndex={openCurrent ? -1 : 0}
          role="presentation"
          className="modal"
          //this is currentTarget
          onClick={this.closeModal}
          onKeyDown={this.closeModal}
          ref={this.modalDomElement}
        >
          <div className="modal-content">{this.props.children}</div>
        </div>
      </React.Fragment>
    );
  }
}

Modal.propTypes = {
  children: PropTypes.any.isRequired,
};

const mapStateToProps = (state, ownProps) => {
  const { open, modalID } = state.modalReducer;
  const { modalID: ownID } = ownProps;
  return {
    open,
    modalID,
    ownID,
  };
};

const mapDispatchToProps = {
  closeModal,
};

export default connect(mapStateToProps, mapDispatchToProps)(Modal);
