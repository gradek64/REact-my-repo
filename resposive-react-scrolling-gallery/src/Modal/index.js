import React from 'react';
import PropTypes from 'prop-types';

// custom imports;
import customEvent from '../utils/customEvents/dispatchEvent';
import './modal.scss';

class Modal extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      open: false,
    };
    this.openModal = this.openModal.bind(this);
    this.closeModal = this.closeModal.bind(this);
  }

  componentDidMount() {
    const { eventOn } = this.props;

    if (!eventOn) {
      console.warn('your custom event is not specified');
      return;
    }
    customEvent.on(eventOn, this.openModal);
    customEvent.on(`CLOSE_${eventOn}`, this.closeModal);
  }

  componentWillUnmount() {
    // only when you change route though;
    // remove custom Event
  }

  openModal() {
    const delaySelector = setTimeout(() => {
      clearTimeout(delaySelector);
      document.querySelector('[tabIndex="-1"]').focus();
    }, 100);
    this.setState({ open: true });
  }

  closeModal(e) {
    if (e && e.target !== e.currentTarget) return;
    if (e.type === 'keydown' && e.key !== 'Escape') return;
    this.setState({ open: false });
  }

  render() {
    const { open } = this.state;
    const { children } = this.props;
    /* eslint-disable no-lone-blocks */
    { /* you want this component to render but not show another approach would be to show on render and
            close on removing component but then dynamic content wont render before showing */ }
    return (
      <React.Fragment>
        {
          <div
            className={open ? 'modal' : 'modal close-state'}
            tabIndex="-1"
            role="presentation"
            onClick={this.closeModal}
            onKeyDown={this.closeModal}
            ref={this.clickAway}
          >
            <div className={open ? 'modal-content slide-in' : 'modal-content'}>
              { children }
            </div>
          </div>
        }
      </React.Fragment>
    );
  }
}

Modal.propTypes = {
  eventOn: PropTypes.string.isRequired,
  children: PropTypes.any.isRequired,
};
export default Modal;
