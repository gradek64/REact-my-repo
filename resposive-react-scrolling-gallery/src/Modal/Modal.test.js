import React from 'react';
import { shallow } from 'enzyme';

import Modal from './index';

const props = {
  eventOn: 'ON_EVENT_ON',
  children: 'that could be any type',
};

const wrapper = shallow(<Modal {...props} />);
test('renders correctly with props given', () => {
  expect(wrapper).toMatchSnapshot();
});

test('it opens and changes state open = true', () => {
  // expect modal to be close initially;
  expect(wrapper.instance().state.open).toBe(false);
  // call class method openModal
  wrapper.instance().openModal();
  // expect to be open now
  expect(wrapper.instance().state.open).toBe(true);
});

test('it closes and changes state open = false', () => {
  // force initial state open=true cause default is false;
  wrapper.setState({ open: true });
  // call class method openModal
  wrapper.instance().closeModal({ type: 'none' });
  // expect to be closed now
  expect(wrapper.instance().state.open).toBe(false);
});

test('it closes on keydown ESC but stays on any other keydown', () => {
  // force initial state open=true cause default is false;
  wrapper.setState({ open: true });

  // call class method openModal
  wrapper.instance().closeModal({ type: 'keydown', key: 'K' });
  // expect to be still open
  expect(wrapper.instance().state.open).toBe(true);

  wrapper.instance().closeModal({ type: 'keydown', key: 'Escape' });
  // expect to be closed
  expect(wrapper.instance().state.open).toBe(false);
});

