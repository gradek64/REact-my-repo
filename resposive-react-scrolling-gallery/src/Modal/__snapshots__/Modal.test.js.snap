// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly with props given 1`] = `
<Fragment>
  <div
    className="modal close-state"
    onClick={[Function]}
    onKeyDown={[Function]}
    role="presentation"
    tabIndex="-1"
  >
    <div
      className="modal-content"
    >
      that could be any type
    </div>
  </div>
</Fragment>
`;
